packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -247,57 +302,63 @@ function getTemplateRoot(
     return root;
 }
 
-function applyHandlers(ctx: ParserCtx, element: IRElement, parsedAttr: ParsedAttribute) {
+function applyHandlers(
+    ctx: ParserCtx,
+    element: Component | Element | Slot,
+    parsedAttr: ParsedAttribute
+) {
     let eventHandlerAttribute;
     while ((eventHandlerAttribute = parsedAttr.pick(EVENT_HANDLER_RE))) {
         const { name } = eventHandlerAttribute;
 
-        if (!isIRExpressionAttribute(eventHandlerAttribute)) {
-            ctx.throwOnIRNode(
+        if (!ir.isExpression(eventHandlerAttribute.value)) {
+            ctx.throwOnNode(
                 ParserDiagnostics.EVENT_HANDLER_SHOULD_BE_EXPRESSION,
                 eventHandlerAttribute
             );
         }
 
         if (!name.match(EVENT_HANDLER_NAME_RE)) {
-            ctx.throwOnIRNode(ParserDiagnostics.INVALID_EVENT_NAME, eventHandlerAttribute, [name]);
+            ctx.throwOnNode(ParserDiagnostics.INVALID_EVENT_NAME, eventHandlerAttribute, [name]);
         }
 
         // Light DOM slots cannot have events because there's no actual `<slot>` element
-        if (element.tag === 'slot' && ctx.getRenderMode(element) === LWCDirectiveRenderMode.light) {
-            ctx.throwOnIRNode(ParserDiagnostics.LWC_LIGHT_SLOT_INVALID_EVENT_LISTENER, element, [
+        if (ir.isSlot(element) && ctx.renderMode === LWCDirectiveRenderMode.Light) {
```

Comment:
Since the slot is parsed before applyHandlers is called, this error is never reached.  There is another error when parsing the slot that checks if there are any invalid directives on the slot.  I'm not sure if we need to keep this or move it further up to when the slot is being parsed.

Reply:
I would move this logic closer to the slot parsing logic now that are treating slots differently than standard elements.

Reply:
I'll add an extra condition when we're initially parsing through the slot